### ISSUE TITLE
`excessive_precision` triggered when adding the `+` in scientific notation

### ISSUE THREAD

**Issue by thi8v** [rust-lang/rust-clippy#15408]
### Summary

Clippy is suggesting to remove the `+` in the scientific notation of a float literal but the suggestion is the same as the code with just more information.

### Lint Name

excessive_precision

### Reproducer

I tried this code:

```rust
pub const MAXIMUM: f64 = 1.797_693_134_862_315_7E+308;
```
[try in playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=b10105c2d13cfd6218ec0256482b296e)

With this example (and maybe others?) clippy's suggestion is
```diff
- pub const MAXIMUM: f64 = 1.797_693_134_862_315_7E+308;
+ pub const MAXIMUM: f64 = 1.797_693_134_862_315_7E308;
```
(so removing the `+`)

I expected to see this happen:

**Nothing**, no warning emitted by clippy.

### Version

```text
rustc 1.88.0 (6b00bc388 2025-06-23)
binary: rustc
commit-hash: 6b00bc3880198600130e1cf62b8f8a93494488cc
commit-date: 2025-06-23
host: x86_64-unknown-linux-gnu
release: 1.88.0
LLVM version: 20.1.5
```

**Comment by samueltardieu** [comment]
This is already fixed in Clippy 1.89 which will be released on Thursday.

Thanks for the report.
